libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -236,35 +230,64 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
 				return undefined;
 			}
 			const appliedOptions = selectCompositeProductAppliedOptions.projector(products, appliedOptionsEntities, { id: props.id });
-			const productDiscountAmount = selectProductDiscountAmount.projector(products, { id: props.id });
-			return daffAdd(productDiscountAmount, getOptionsDiscountAmount(<DaffCompositeProduct>product, appliedOptions));
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
+				acc, 
+				appliedOptions[item.id] ? 
+					daffMultiply(
+						daffSubtract(appliedOptions[item.id].price, appliedOptions[item.id].discount && appliedOptions[item.id].discount.amount), 
```

Comment:
I think the null check should be handled like:
```ts
appliedOptions[item.id].discount ? daffSubtract(appliedOptions[item.id].price, appliedOptions[item.id].discount.amount) : appliedOptions[item.id].price
```
to avoid running `daffSubtract` any more than necessary.

Reply:
changed